spring/src/main/java/org/axonframework/spring/config/annotation/SpringBeanParameterResolverFactory.java

```diff
@@ -76,16 +81,26 @@ public ParameterResolver createInstance(Executable executable, Parameter[] param
         } else if (beansFound.size() > 1) {
             final AutowireCapableBeanFactory beanFactory = applicationContext.getAutowireCapableBeanFactory();
             if (beanFactory instanceof ConfigurableListableBeanFactory) {
+                final ConfigurableListableBeanFactory clBeanFactory = (ConfigurableListableBeanFactory) beanFactory;
+                // find @Qualifier matching candidate
+                final Optional<Map<String, Object>> qualifier = AnnotationUtils.findAnnotationAttributes(parameters[parameterIndex], Qualifier.class);
```

Comment:
Small suggestion, but might be nice to extract this as a separate private function (as well as the `@Primary` candidate check), so that we have something like `findBeanBasedOnQualifier(parameters, , parameterIndex, beanFactory)`